### PR TITLE
make bazel scenario automatically exclude manual targets

### PR DESCRIPTION
**Pull request by BenTheElder**
[bazel actually does this in their CI scripts](https://bazel-review.googlesource.com/c/bazel/+/1800/4/scripts/ci/ci.sh)

This change means we don't have to manually exclude manual targets in our build/test target patterns in CI, when we query we will automatically exclude them, however in case we want to go ahead and run some manual target, I added new flags `--manual-build` and `--manual-test` to support this.

### CODE REVIEW

**Comment by BenTheElder** [comment]
Current manual targets:
```
$ cd $GOPATH/src/k8s.io/kubernetes && git-sync-upstream-master && bazel query 'attr('tags', 'manual', //...)'
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 7d4f21d0d7 Merge pull request #54298 from mwielgus/ca-1.0.1
Everything up-to-date
//hack:verify-all
//hack:verify-boilerplate
//examples:go_default_xtest
//cluster/images/kubemark:push
//build/release-tars:package_src
//:push-build
```
```
$ cd $GOPATH/src/k8s.io/test-infra && git-sync-upstream-master && bazel query 'attr('tags', 'manual', //...)'
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 53826b4c Merge pull request #5102 from krzyzacy/fix-registry
Everything up-to-date
//metrics:k8
//metrics:darwin
```

File: prow/config.yaml [hunk 1]
```diff
@@ -3460,6 +3460,46 @@ postsubmits:
         secret:
           secretName: service-account
 
+  kubernetes/cluster-registry:
```
Comment (ixdy) [inline prow/config.yaml]: was this intentional?

Comment (BenTheElder) [reply prow/config.yaml]: no ><

Comment (BenTheElder) [reply prow/config.yaml]: removed

File: prow/config.yaml [hunk 2]
```diff
@@ -505,7 +505,7 @@ presubmits:
         - "--service-account=/etc/service-account/service-account.json"
         - "--upload=gs://kubernetes-jenkins/pr-logs"
         - "--" # end bootstrap args, scenario args below
-        - "--test=//... //hack:verify-all -//build/... -//vendor/... -//examples/..."
+        - "--test=//... //hack:verify-all -//build/... -//vendor/..."
```
Comment (ixdy) [inline prow/config.yaml]: I think `//hack:verify-all` needs to be a `--manual-test` now?

Comment (BenTheElder) [reply prow/config.yaml]: yes. I should query the other branches for manual as well.

**Comment by BenTheElder** [comment]
/hold
querying each release

**Comment by BenTheElder** [comment]
release-1.6:
```
//hack:verify-all
//hack:verify-gofmt
//hack:verify-boilerplate
//build/release-tars:package_src
```

release-1.7:
```
//vendor/github.com/seccomp/libseccomp-golang:cgo_codegen
//vendor/github.com/opencontainers/runc/libcontainer/system:cgo_codegen
//vendor/github.com/google/cadvisor/utils/cpuload/netlink:cgo_codegen
//vendor/github.com/docker/docker/pkg/term:cgo_codegen
//vendor/github.com/coreos/pkg/dlopen:cgo_codegen
//vendor/github.com/coreos/go-systemd/util:cgo_codegen
//pkg/kubelet/eviction:cgo_codegen
//hack:verify-all
//hack:verify-boilerplate
//build/release-tars:package_src
```

release-1.8:
```
//hack:verify-all
//hack:verify-boilerplate
//build/release-tars:package_src
```

File: prow/config.yaml (hunk 2 above)
Comment (BenTheElder) [reply prow/config.yaml]: moved `//hack:verify-all` for each branch, I think test is fine now, inspecting build.

**Pull request merged**